v1.9 backports 2021-04-27

nebril opened the pull request:
* #14895 -- bugtool: Record attached BPF programs (@aditighag)
 * #15762 -- test: Skip K8sPolicy on GKE and 4.19 (@pchaigno)
 * #15838 -- contrib: Clean output of submit-backport script (@pchaigno)
 * #15841 -- .github: remove unnecessary docker hub credentials (@aanm)
 * #15836 -- docs/policy: Clarify table for deny policy scenarios (@pchaigno)
 * #15863 -- test/gke: refactor test-clusters operations (@nbusseneau)

Once this PR is merged, you can update the PR labels via:
```upstream-prs
$ for pr in 14895 15762 15838 15841 15836 15863 ; do contrib/backporting/set-labels.py $pr done 1.9; done
```

aditighag committed:
bugtool: Record attached BPF programs

[ upstream commit 0412be84ca00c3764b908cc80d036ea1addf5635 ]

bpftool net show command lists all the attached programs.
We sometimes need to get a summary of BPF programs/SECs
attached to different interfaces to debug datapath related issues.

Sample output -

bpftool net show
xdp:

tc:
eth0(2) clsact/ingress bpf_network.o:[from-network] id 221
cilium_net(4) clsact/ingress bpf_host_cilium_net.o:[to-host] id 230
cilium_host(5) clsact/ingress bpf_host.o:[to-host] id 222
cilium_host(5) clsact/egress bpf_host.o:[from-host] id 226
lxc_health(7) clsact/ingress bpf_lxc.o:[from-container] id 242
lxc_health(7) clsact/egress bpf_lxc.o:[to-container] id 246
lxc37bff36e0cd2(9) clsact/ingress bpf_lxc.o:[from-container] id 234
lxc37bff36e0cd2(9) clsact/egress bpf_lxc.o:[to-container] id 238
lxc2686a607b303(12) clsact/ingress bpf_lxc.o:[from-container] id 250
lxc2686a607b303(12) clsact/egress bpf_lxc.o:[to-container] id 254
lxc33d2535b88de(14) clsact/ingress bpf_lxc.o:[from-container] id 258
lxc33d2535b88de(14) clsact/egress bpf_lxc.o:[to-container] id 262

flow_dissector:

Signed-off-by: Aditi Ghag <aditi@cilium.io>
Signed-off-by: Maciej Kwiek <maciej@isovalent.com>

pchaigno committed:
test: Skip K8sPolicies on GKE and 4.19

[ upstream commit 1fff6e98487956da9e6c37f83b6b82c7b88f9ec9 ]

Running K8sPolicies on those CI jobs is not expected to increase
coverage, so let's disable to reduce cost.

We need to skip in both BeforeAll and JustBeforeEach because we want to
skip the installation in BeforeAll but doing so only skips the first
test in K8sPolicyTest. So we also want to skip in JustBeforeEach.

Signed-off-by: Paul Chaignon <paul@cilium.io>
Signed-off-by: Maciej Kwiek <maciej@isovalent.com>

pchaigno committed:
docs/policy: Clarify table for deny policy scenarios

[ upstream commit a6eba34564e4cb23d052e2dee7ac9ee510f5f62e ]

This commit attempts to clarify the table detailed different deny policy
scenarios. The titles are changed to clarify that the bottom part gives
the result from each scenario detailed in the top part. Yes/No are
replaced by Xs as that makes the table a bit easier to parse visually
and probably also better represents the presence/absence of policies.

Signed-off-by: Paul Chaignon <paul@cilium.io>
Signed-off-by: Maciej Kwiek <maciej@isovalent.com>

pchaigno committed:
contrib: Clean output of submit-backport

[ upstream commit b1221c36739d4720025b3a3718848d82d269c876 ]

When using submit-backport to open the backport pull request, it outputs
the following information:

    Sending pull request...
    Enumerating objects: 154, done.
    Counting objects: 100% (115/115), done.
    Delta compression using up to 6 threads
    Compressing objects: 100% (24/24), done.
    Writing objects: 100% (24/24), 3.27 KiB | 3.27 MiB/s, done.
    Total 24 (delta 20), reused 0 (delta 0)
    remote: Resolving deltas: 100% (20/20), completed with 12 local objects.
    remote:
    remote: Create a pull request for 'pr/v1.9-backport-2021-04-22' on GitHub by visiting:
    remote:      https://github.com/pchaigno/cilium/pull/new/pr/v1.9-backport-2021-04-22
    remote:
    To github.com:pchaigno/cilium
     * [new branch]          pr/v1.9-backport-2021-04-22 -> pr/v1.9-backport-2021-04-22
    https://github.com/cilium/cilium/pull/15837

    Updating labels for PRs 15780

This is a bit messy and mostly caused by the git push command. We can
hide most of this since the script is in charge of opening the pull
request.

Signed-off-by: Paul Chaignon <paul@cilium.io>
Signed-off-by: Maciej Kwiek <maciej@isovalent.com>

nbusseneau committed:
test/gke: remove obsolete script

[ upstream commit 44f12e3a2ed62a8d1985e05337b2389fc0ff1e8c ]

Signed-off-by: Nicolas Busseneau <nicolas@isovalent.com>
Signed-off-by: Maciej Kwiek <maciej@isovalent.com>

nbusseneau committed:
test/gke: refactor gcloud container operations

[ upstream commit 635b9070a9539cd8cdf783e1077b9e272b19a37f ]

Using server-side filtering greatly accelerates checking for running
operations.

Signed-off-by: Nicolas Busseneau <nicolas@isovalent.com>
Signed-off-by: Maciej Kwiek <maciej@isovalent.com>

nbusseneau committed:
test/gke: handle operations via ConfigConnector only

[ upstream commit 9e21ed118e0d91cb7468f5aacea77855d9f3ca96 ]

ConfigConnector already takes care of resizing/deletion for us, no need for manually handling that.

Signed-off-by: Nicolas Busseneau <nicolas@isovalent.com>
Signed-off-by: Maciej Kwiek <maciej@isovalent.com>

aanm reviewed (approved):
Looks good for my change

aditighag reviewed (approved):
>#14895 -- bugtool: Record attached BPF programs (@aditighag)

lgtm

nbusseneau committed:
test/gke: fix retry loop failing

[ upstream commit f0a4292c2354f9eb5ebee928eed8a221041de6e0 ]

From `bash` manpage:

> ((expression))
> The expression is evaluated according to the rules described below
> under ARITHMETIC EVALUATION.  If the value of the expression is
> non-zero, the return status is 0; otherwise the return status is 1.
> This is exactly equivalent to let "expression".

With `set -o errexit`, this means `((resize_wait_retries++))` fails...

Signed-off-by: Nicolas Busseneau <nicolas@isovalent.com>
Signed-off-by: Maciej Kwiek <maciej@isovalent.com>

pchaigno reviewed (approved):
My changes look good :+1:

nbusseneau commented on Documentation/contributing/testing/e2e.rst:
```diff
@@ -503,9 +503,9 @@ cluster.
-          and can get rid of them by running ``cilium/test/gke/clean-cluster.sh``.
+.. note:: The tests require the ``NATIVE_CIDR`` environment variable to be set to
+          the value of the cluster IPv4 CIDR returned by the ``gcloud container
+          clusters describe`` command.
```
The original commit only deleted the part on the left, not sure why this bit got added in.

nebril replied:
Will delete, I guess I messed up the conflict resolution.

nebril committed:
.github: remove unnecessary docker hub credentials

[ upstream commit 31625355bec914998bbf372a0519a87a4a8e1c3f ]

GHA are not rate limited by DockerHub [1] so we can remove these
credentials

[1] https://github.com/actions/virtual-environments/issues/1445#issuecomment-713861495

Signed-off-by: André Martins <andre@cilium.io>
Signed-off-by: Maciej Kwiek <maciej@isovalent.com>
